src/pond_soil_model.jl

```diff
@@ -242,5 +233,5 @@ function Soil.boundary_fluxes(
     p::ClimaCore.Fields.FieldVector,
     t::FT,
 ) where {FT}
-    return p.soil_infiltration[1], FT(0.0)
+    return p.soil_infiltration[], FT(0.0)
```

Comment:
You can access the values in a `ClimaCore.DataLayouts.DataF` (the backing datatype for the `PointField`) with `[]`. You will get `getindex` bounds errors otherwise.

Reply:
Ah OK. So, the boundary_fluxes method for the soil model returns floats. I guess we can broadcast these methods over fields (and this would still pull out the "float" object correctly) or we can still apply at each column (and it would do the same, since column (planar surface) =Point).

This looks fine for now!

Reply:
I need to play with this, but actually you make a good point that we shouldn't need to call into the DataLayout here, we can just return the `PointField`.

Reply:
This is resolved - see the changes to `rre.jl`